mono/mini/interp/transform.c

```diff
@@ -1196,7 +1196,31 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoMeth
 			td->ip += 5;
 			return TRUE;
 		}
+	} else if (in_corlib && !strcmp (klass_name_space, "Internal.Runtime.CompilerServices") && !strcmp (klass_name, "Unsafe")) {
+#ifdef ENABLE_NETCORE
+		if (!strcmp (tm, "AddByteOffset"))
+			*op = MINT_INTRINS_UNSAFE_ADD_BYTE_OFFSET;
+		if (!strcmp (tm, "ByteOffset"))
```

Comment:
`else if`

Reply:
It is too bad the strings don't carry their lengths with them, in an efficient to get way.